examples/cpp/raw_mandelbrot.cpp

```diff
@@ -0,0 +1,175 @@
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <new>
+
+#include "pf.h"
+
+#include "model/colormap.h"
+#include "model/image.h"
+#include "model/enums.h"
+#include "model/imagewriter.h"
+
+#include "model/fractfunc.h"
+#include "model/vectors.h"
+
+
+int main() {
+    // initial setup: load fract4_stdlib globally so the loaded formula has access to it
+    void *fract_stdlib_handle = dlopen("./fract_stdlib.so", RTLD_GLOBAL | RTLD_NOW);
+    if (NULL == fract_stdlib_handle) {
+        fprintf(stderr, "Error loading libfract_stdlib: %s", dlerror());
+        return -1;
+    }
+
+    // load formula lib
+    void *lib_handle = dlopen("./formula.so", RTLD_NOW);
+    if (NULL == lib_handle)
+    {
+        fprintf(stderr, "Error loading formula: %s", dlerror());
+        return -1;
+    }
+    pf_obj *(*pfn)(void);
+    pfn = (pf_obj * (*)(void)) dlsym(lib_handle, "pf_new");
+    if (NULL == pfn)
+    {
+        fprintf(stderr, "Error loading formula symbols: %s", dlerror());
+        dlclose(lib_handle);
+        return -1;
+    }
+    pf_obj *pf_handle = pfn();
+
+    // formula params: [0, 4.0, 0.0, 1.0, 4.0, 0.0, 1.0]
+    int param_len = 7;
+    struct s_param *params = (struct s_param *)malloc(param_len * sizeof(struct s_param));
+    params[0].t = INT;
+    params[0].intval = 0;
+    params[1].t = FLOAT;
+    params[1].doubleval = 4.0;
+    params[2].t = FLOAT;
+    params[2].doubleval = 0.0;
+    params[3].t = FLOAT;
+    params[3].doubleval = 1.0;
+    params[4].t = FLOAT;
+    params[4].doubleval = 4.0;
+    params[5].t = FLOAT;
+    params[5].doubleval = 0.0;
+    params[6].t = FLOAT;
+    params[6].doubleval = 1.0;
+    // position params
+    double pos_params[N_PARAMS] {
```

Comment:
This could be constexpr, which would make this compile-time or at least automatically const static

Reply:
If I make it constant then I had to cast it to non-const to use it in `pf_handle->vtbl->init` (with const_cast for example).
Would that make it any better?

Reply:
ah, yeah.. avoid const_cast<>() - it's a bastion of instant undefined behaviour. I would still make it static so you don't pay for constructing it every call.

This problem suggests the vtbl interface needs redesigning for const-correctness as that stinks of bugs (a code-smell as lefticus would say)

Reply:
I'm working on an example that includes a custom formula code. That's something to have in mind.
To apply that improvement to gnofract we need to change the compiler, I think it shouldn't be very complicated (I guess that is part of a literal template of the C code generated for the formula).

Reply:
That works. And yeah, from what I recall about the code gen, it's simple enough to change this.